zpool/zfs list fixes

nabijaczleweli opened the pull request:
### Motivation and Context
#13122, #13123; I wrote them down to not forget and then I did them before I'd've forgotten! what's the point

### Description
See individual commit messages.

### How Has This Been Tested?
Likewise.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes. – none apply
- [ ] I have run the ZFS Test Suite with this change applied. – CI take my hand
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

behlendorf commented on include/zfs_prop.h:
```diff
@@ -76,10 +76,11 @@ typedef struct {
 					/* "zfs get" help message */
-	boolean_t pd_zfs_mod_supported;	/* supported by running zfs module */
+	boolean_t pd_zfs_mod_supported: 1; /* supported by running zfs module */
+	boolean_t pd_always_flex: 1;	/* never fixed-width */
```
It feels a little out of place to make this optimization to this one structure in this PR.  If we want to tackle this I'd suggest we look in to doing something a bit more comprehensive, or perhaps targeting the structures which will have the biggest payoff.

nabijaczleweli replied:
I mean, I'm already adding a field and breaking other ABIs so I've decided that might as well, and I'm gonna guess that across all the properties (4*sizeof(int)-1)*(95+33+41)=2535 bytes saved in .bss is, like, *definitely* worthwhile?

I was gonna update the ABI after my other libzfs-ABI-touching PR landed, but I can do it now and rebase that one later, instead, if you prefer pulling this one in first?

behlendorf replied:
Fair enough, it does all add up.

behlendorf reviewed (approved):
Looks good.

behlendorf commented:
It appears there's still a discrepancy with the ABI, https://github.com/openzfs/zfs/runs/5368831982?check_suite_focus=true

nabijaczleweli commented:
Yeah, I rebased past 361a7e821178e105c8e1206ead4479de83c2a617 which included a different ABI break; updated with latest ABI off the CI, should be fine now

nabijaczleweli commented:
Aaaand it passes. What joy.
